scripts/cji_smoke_test.sh

```diff
@@ -0,0 +1,228 @@
+#IQE_RP_ARGS=True -- Turn on reporting to reportportal
+#IQE_IBUTSU_SOURCE="post_stage" -- update the ibutsu source for the current run
+#IQE_ENV_VARS="ENV_VAR1=value1,ENV_VAR2=value2" -- custom set of extra environment variables to set on IQE pod
+#NAMESPACE="mynamespace" -- namespace to deploy iqe pod into, usually already set by 'deploy_ephemeral_env.sh'
+
+# Env vars set by 'bootstrap.sh':
+#ARTIFACTS_DIR -- directory where test run artifacts are stored
+
+# In order for the deploy-iqe-cji to run correctly, we must set the marker and filter to "" if they
+# are not already set by caller
+# https://unix.stackexchange.com/questions/122845/using-a-b-for-variable-assignment-in-scripts/122848#122848
+set -e
+
+: "${IQE_MARKER_EXPRESSION:='""'}"
+: "${IQE_FILTER_EXPRESSION:='""'}"
+: "${IQE_IMAGE_TAG:='""'}"
+: "${IQE_REQUIREMENTS:='""'}"
+: "${IQE_REQUIREMENTS_PRIORITY:='""'}"
+: "${IQE_TEST_IMPORTANCE:='""'}"
+: "${IQE_PLUGINS:='""'}"
+: "${IQE_ENV:=clowder_smoke}"
+: "${IQE_SELENIUM:=false}"
+: "${IQE_PARALLEL_ENABLED:='""'}"
+: "${IQE_PARALLEL_WORKER_COUNT:='""'}"
+: "${IQE_RP_ARGS:='""'}"
+: "${IQE_IBUTSU_SOURCE:='""'}"
+: "${IQE_ENV_VARS:=}"
+
+_running_in_rhel7() {
+    grep -q "Red Hat Enterprise Linux.*7\." '/etc/redhat-release'
+}
+
+# REMOVE after https://issues.redhat.com/browse/APPSRE-7886
+if _running_in_rhel7; then
+    MC_IMAGE="quay.io/cloudservices/mc:RELEASE.2023-05-30T22-41-38Z"
+else
+    MC_IMAGE="quay.io/cloudservices/mc:latest"
+fi
+
+# minio client is used to fetch test artifacts from minio in the ephemeral ns
+echo "Running: docker pull ${MC_IMAGE}"
+docker pull ${MC_IMAGE}
+
+if [[ -z $IQE_CJI_TIMEOUT ]]; then
+    echo "Error: no timeout set; export IQE_CJI_TIMEOUT before invoking cji_smoke_test.sh"
+    exit 1
+fi
+
+SELENIUM_ARG=""
+if [ "$IQE_SELENIUM" = "true" ]; then
+    SELENIUM_ARG=" --selenium "
+fi
+
+ENV_VAR_ARGS=""
+if [ ! -z "$IQE_ENV_VARS" ]; then
+    IFS=',' read -ra values_array <<< "$IQE_ENV_VARS"
+    for i in "${values_array[@]}"; do
+        ENV_VAR_ARGS="${ENV_VAR_ARGS} --env-var $i"
+    done
+fi
+
+# check if there is a iqe-tests container image tag with the corresponding PR
+# number if not use the default IQE_IMAGE_TAG tag which you have to set in
+# order for this to work properly
+if [ ! -z "$ghprbPullId" ]; then
+    export IQE_IMAGE_TAG_TMP="${IQE_IMAGE_TAG}-pr-${ghprbPullId}"
+fi
+
+if [ ! -z "$gitlabMergeRequestIid" ]; then
+    export IQE_IMAGE_TAG_TMP="${IQE_IMAGE_TAG}-pr-${gitlabMergeRequestIid}"
+fi
+
+if [ ! -z "$IQE_IMAGE_TAG_TMP" ]; then
+    set +e
+    if docker manifest inspect quay.io/cloudservices/iqe-tests:"$IQE_IMAGE_TAG_TMP"; then
+        echo "Found IQE_IMAGE_TAG=$IQE_IMAGE_TAG_TMP. It will be used for testing."
+        export IQE_IMAGE_TAG="$IQE_IMAGE_TAG_TMP"
+    fi
+    set -e
+fi
+
+# Set variables that didn't resolve properly
+CJI_NAME="runtimes-inventory-smokes"
+IQE_IMAGE_TAG="runtimes-inventory"
```

Comment:
`IQE_IMAGE_TAG` should work out of the box if the plugin name is specified [in our clowdapp.yml](https://github.com/RedHatInsights/insights-runtimes-inventory/blob/main/deploy/clowdapp.yml#L134)
Having it hard coded like this makes it impossible to pass your own custom tag (to eg run with other plugin versions).

And I'm not sure why we need the custom CJI_NAME? Was there an issue with the original `CJI_NAME="$COMPONENT_NAME"` ?

I'm asking before AFAICT those are the only two changes compared to the original cji_smoke_test, so if we could drop this, we could get rid of the script completely. I'd rather not have to maintain our own copy :)

Reply:
If I'm reading the iqe plugin documentation correctly, isn't the cji-name supposed to be `runtime-inventory-smokes`? In the script from cicd it sets the cji_name to be the component_name, which would be `runtime-inventory`, and it was unable to find it.

Reply:
>  In the script from cicd it sets the cji_name to be the component_name, which would be runtime-inventory, and it was unable to find it.

The relevant part for finding the test image should be `--image-tag "${IQE_IMAGE_TAG}"` that references the `quay.io/cloudservices/iqe-tests` image tag that should be used. So I think that was also caused by the empty iqePlugin in our clowdapp. Sorry about that one I meant to update it but there was a bug somewhere in bonfire / clowder (fixed since then) when I was dealing with CJI so I couldn't test it and then just forgot about it :(